### ISSUE TITLE
EmailField showing invalid when empty and not required

### ISSUE THREAD

**Issue by drewharvey** [vaadin/flow-components#5361]
### Description of the bug

Vaadin 23.3.19 with `enforceFieldValidation=true`

EmailField will be marked as invalid if you enter text and then clear the text.  It will only show as valid if you enter an email.

![emailbug](https://github.com/vaadin/flow/assets/2496955/a4e07e4f-aa6a-4987-ad2b-d960e400dec9)

**Repro steps:**
- Enter text into email field and blur
- Clear text from email field and blur
- Expected: field shown as valid and empty
- Actual: field shown as invalid and empty

### Expected behavior

Field should be valid if empty.

### Minimal reproducible example

var email = new EmailField("Email");
add(email);

### Versions

- Vaadin / Flow version: 23.3.19
com.vaadin.experimental.enforceFieldValidation=true

**Comment by drewharvey** [comment]
In case anyone needs a workaround, you can override the `isFeatureFlagEnabled` method and force the field to behave as if the `ENFORCE_FIELD_VALIDATION` flag is false.

```
var field = new EmailField("Email") {
    @Override
    public boolean isFeatureFlagEnabled(Feature feature) {
        if (feature == FeatureFlags.ENFORCE_FIELD_VALIDATION)
            return false;
        return super.isFeatureFlagEnabled(feature);
    }
};
```

**Comment by drewharvey** [comment]
Seems all custom validators do not run until a valid email formatted string is entered.  You can see in the GIF posted above that when using `enforceFieldValidation` the EmailValidator does not trigger.

**Comment by vursen** [comment]
> Seems all custom validators do not run until a valid email formatted string is entered.

Yeah, it's because Binder first runs the default validator provided by the component. And the email validation is part of that default validator. However, there is a known problem that you cannot customize error messages for errors resulting from the default validator. This is why the field appears red without any error message. Please, refer to #4618 for further updates.
